### ISSUE TITLE
Unable to use @Id mapping together with a Grid bound to a specific bean class

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow-components#847]
What you would want to do is that having this in the template

```html
<vaadin-grid id="grid"></vaadin-grid>
```

and this in Java

```
public class MyTemplate extends LitTemplate {

@Id
private Grid<Person> grid;
```

would allow you to get a `Grid` instance that is bound to `Person.class`. The same you would get with
```
Grid<Person> grid = new Grid<>(Person.class)
```

This would allow two things:
1. Adding all columns based on `Person`
2. Using `addColumn("something")` or `getColumn("something")` to customize columns

Right now you have to define all columns manually using the getter syntax.

**Comment by Artur-** [comment]
There are multiple ways to solve this but what feels like the easiest would be to allow setting the Grid bean type after the Grid instance has been created, so in the case above you could do
```
grid.setBeanType(Person.class);
```
in the constructor

**Comment by sissbruecker** [comment]
Assigned to @rolfsmeds  to declare AC.

**Comment by rolfsmeds** [comment]
## Description
Bean type configuration API for Grid that allows setting the bean type after creating a new instance.

## Use cases
As a developer
I want to define the bean class for a Grid after instantiating it
So that I don't have to know the bean type at the time of instantiation

As a developer
I want to define a bean class for an @Id-mapped Grid in a LitTemplate
So that I can manage the Grid's columns easily in Java

## Acceptance criteria
- [ ] A method for configuring a Grid's bean type separately from the constructor...
- [ ] ...that allows the developer to choose whether to also create corresponding columns automatically
- [ ] This API can only be used on Grids without a beantype or columns.
- [ ] If the Grid already has a beantype or one or more columns, using this API should throw an exception.

## General criteria

- [ ] UX/DX tests in Alpha 


- [ ] Documentation: 


- [ ] How to test? 


- [ ] Limitations:

**Comment by sissbruecker** [comment]
Fixed as part of #600
